Add unit tests for Store/AppAccess and Store/UserAccess

DaloonOfDoom opened the pull request.

DaloonOfDoom committed:
Add unit tests for Store/AppAccess and Store/UserAccess

Omarley7 commented on client/test/unitTests/Store/UserAccess.test.tsx:
```diff
@@ -0,0 +1,20 @@
+  const defaultRender = wrapWithInitialState();
+  const testUserName = 'testUser';
+
+  it('should update user correctly with setUser', () => {
```
I recommend extention of testUtil. We need something like
```ts
export function renderHookWithInitialState(hook: function) {
  return renderHook(() => hook, {
      wrapper: defaultRender,
    });
}
```

Omarley7 replied:
Could it make sense to give optional state extension? 
```ts
export function renderHookWithInitialState(hook: function, initState?)
```

This will also invite refactoring of other uses of the wrapWithInitialState() function.

Omarley7 commented on client/test/unitTests/Store/AppAccess.test.tsx:
```diff
@@ -0,0 +1,38 @@
+  const defaultRender = wrapWithInitialState();
+
+  it('menu should be closed initially', () => {
+    const { result } = renderHook(() => useApp(), { wrapper: defaultRender });
```
Should reuse a test setup functino

Omarley7 commented on client/test/unitTests/Store/AppAccess.test.tsx:
```diff
@@ -0,0 +1,38 @@
+  });
+
+  it('menu should open correctly', () => {
+    const { result } = renderHook(() => useApp(), { wrapper: defaultRender });
```
test setup function needed

Omarley7 reviewed (changes requested):
Codeclimate issues, but otherwise nice. We need to be very modular with the codebase

Omarley7 committed:
Honestly bullshit codeclimate error.
Would be overly complicated to fix. Even this solution is stupid.
And also updated envUtil to use the same hook, act, assert approach.

Omarley7 committed:
Bullshit løsning
